Documentation/contributor-guide/exit_codes.md

```diff
@@ -0,0 +1,52 @@
+# Exit Codes Reference
+
+This document provides a reference of exit codes returned by the etcd server.
+
+etcd server explicitly uses three exit codes: **0** (success), **1** (general errors), and **2** (argument errors). When terminated by signals (SIGTERM/SIGINT), the exit code depends on the process type: PID 1 processes exit with code 0, while non-PID 1 processes re-raise the signal, resulting in exit codes 143 (SIGTERM) or 130 (SIGINT).
+
+## Exit Code 0 - Success
+
+| Scenario | Code Reference |
+| -------- | -------------- |
+| Help flag (`--help`) | [`server/etcdmain/config.go#L131`](https://github.com/etcd-io/etcd/blob/e0a72cf470756149f4f602bf89284038e6397549/server/etcdmain/config.go#L131) |
+| Version flag (`--version`) | [`server/etcdmain/config.go#L144`](https://github.com/etcd-io/etcd/blob/e0a72cf470756149f4f602bf89284038e6397549/server/etcdmain/config.go#L144) |
+| Normal shutdown | [`server/etcdmain/etcd.go#L176`](https://github.com/etcd-io/etcd/blob/e0a72cf470756149f4f602bf89284038e6397549/server/etcdmain/etcd.go#L176) |
+| Graceful shutdown on signal (PID 1) | [`pkg/osutil/interrupt_unix.go#L74`](https://github.com/etcd-io/etcd/blob/e0a72cf470756149f4f602bf89284038e6397549/pkg/osutil/interrupt_unix.go#L74) |
+
+## Signal Termination (128 + signal number)
+
+For non-PID 1 processes, the signal handler re-raises the signal to the process itself ([`pkg/osutil/interrupt_unix.go#L77`](https://github.com/etcd-io/etcd/blob/e0a72cf470756149f4f602bf89284038e6397549/pkg/osutil/interrupt_unix.go#L77)), which results in the kernel setting the exit code to 128 + signal number.
```

Comment:
Please clarify that this behavior is specific to Linux. On all other platforms, etcd typically returns 0 if it exits without error, and 1 otherwise.

Reply:
FYI.

https://github.com/etcd-io/etcd/blob/aef0361938cbc604fef2ac15996518988d0c9290/pkg/osutil/signal_linux.go#L15

https://github.com/etcd-io/etcd/blob/aef0361938cbc604fef2ac15996518988d0c9290/pkg/osutil/signal.go#L15


For example, on macOS, the exit code is 0 when etcd is terminated by signals